### PR TITLE
[Documentation]: Documenting the GitHubFlow workflow and create sequence diagrams

### LINKED ISSUE #4063: [Documentation]: Documenting the GitHubFlow workflow and create sequence
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched [issues](https://github.com/GitTools/GitVersion/issues) to ensure it has not already been reported

### GitVersion package

GitVersion.Tool

### GitVersion version

6.0.0-rc.1

### Operating system

N/A

### What are you seeing?

Documenting the GitHubFlow workflow and create sequence diagrams

### What is expected?

The GitHubFlow workflow is documented

### Steps to Reproduce

n/a

### RepositoryFixture Test

_No response_

### Output log or link to your CI build (if appropriate).

_No response_

### PR DESCRIPTION
**Pull request by HHobeck**
## Description
- #4063

Close #4063

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by arturcic** [comment]
@asbjornu can you review the PR? thanks

**Comment by HHobeck** [comment]
@arturcic please merge this. Only the documentation has changed

File: docs/input/docs/learn/branching-strategies/gitflow/examples.md [hunk 1]
```diff
@@ -86,6 +86,15 @@ iterations. Release branches are taken from `main` (or from `develop`) and will
 be merged back afterwards. Finally the `main` branch is tagged with the
 released version.
 
+Release branches can be used in the `GitFlow` as well as `GitHubFlow` workflow.
+Sometimes you
```
Comment (arturcic) [inline docs/input/docs/learn/branching-strategies/gitflow/examples.md:90]: I think we can remove the newline here

**Review by arturcic — CHANGES_REQUESTED** [review]

File: docs/input/docs/learn/branching-strategies/gitflow/examples.md [hunk 2]
```diff
@@ -86,6 +86,15 @@ iterations. Release branches are taken from `main` (or from `develop`) and will
...
+want to start on a large feature which may take a while to stabilize so you want
+to keep it off main. In these scenarios you can either create a long lived
```
Comment (arturcic) [inline docs/input/docs/learn/branching-strategies/gitflow/examples.md:92]:
```suggestion
Sometimes you want to start on a large feature which may take a while 
to stabilize so you want to keep it off main. 
In these scenarios you can either create a long lived
```

**Review by arturcic — APPROVED** [review]

**Pull request merged**